apm-agent-core/src/main/java/co/elastic/apm/agent/impl/GlobalTracer.java

```diff
@@ -54,17 +63,52 @@ public static ElasticApmTracer requireTracerImpl() {
         return Objects.requireNonNull(getTracerImpl(), "Registered tracer is not an instance of ElasticApmTracer");
     }
 
+    private static void checkClassloader() {
+        ClassLoader cl = GlobalTracer.class.getClassLoader();
+
+        if (classloaderCheck(cl)) {
+            return;
+        }
+
+        throw new IllegalStateException("Agent setup error: must be loaded in bootstap classloader, should not be included in application dependencies. Classloader : " + cl);
```

Comment:
[proposal] I think this message should be very focused on how the user can fix it - no need to say anything about class loaders even, only that the agent seems to have been setup improperly, and it is likely referenced by the JVM or application classpath, or used as a dependency.

Reply:
Definitely agree here, the reason people tend to put this `.jar` on the classpath is because they have heard too much "put things on classpath" :-).